src/wechaty_puppet/memory_card/memory_card.py

```diff
@@ -0,0 +1,332 @@
+"""doc"""
+from __future__ import annotations
+from os import path
+import logging
+import json
+import re
+from typing import (
+    Optional,
+    List,
+    Union,
+    Any
+)
+
+from dataclasses import dataclass
+
+from .config import VERSION
+from .storage import (
+    getStorage,
+    StorageBackend,
+    StorageBackendOptions
+)
+from .types import (
+    AsyncMap,
+    MemoryCardPayload
+)
+import collections
+log = logging.getLogger('memory_card')
+NAMESPACE_MULTIPLEX_SEPRATOR = '\r'
+NAMESPACE_KEY_SEPRATOR = '\n'
+# TODO?
+NAMESPACE_MULTIPLEX_SEPRATOR_REGEX = re.compile(NAMESPACE_MULTIPLEX_SEPRATOR)
+NAMESPACE_KEY_SEPRATOR_REGEX = re.compile(NAMESPACE_KEY_SEPRATOR)
+
+
+@dataclass
+class multiplexprop:
+    parent: MemoryCard
+    name: str
+
+
+@dataclass
+class MemoryCardOptions:
+    # TODO name: Optional[str]
+    name: Optional[str] = None
+    storageOptions: Optional[StorageBackendOptions] = None
+    multiplex: Optional[multiplexprop] = None
+
+
+@dataclass
+class MemoryCardJsonObject:
+    payload: MemoryCardPayload
+    options: MemoryCardOptions
+
+
+# TODO
+class MemoryCard(AsyncMap):
+    VERSION = VERSION
+
+    @staticmethod
+    def fromJSON(textOrObj) -> MemoryCard:
+        log.info('MemoryCard', 'fromJSON(...)')
+        jsonObj: MemoryCardJsonObject
+        if type(textOrObj) == str:
+            jsonObj = json.loads(textOrObj)
+        else:
+            jsonObj = textOrObj
+        card = MemoryCard(jsonObj.options)
+        return card
+
+    # @classmethod
+    # def multiplex(cls, memory: MemoryCard, name: str) -> MemoryCard:
+    #     log.info('MemoryCard', 'static multiplex(%s, %s)' % (memory, name))
+    #     mpMemory = cls(memory.options)
+    #     return mpMemory
```

Comment:
Is it used in ts?But it is redeclared in python.

Reply:
I think it's a interface which can be extended to a more flexible component.  Leaving it here is fine.